src/common/operator/string_cast.cpp

```diff
@@ -159,6 +159,55 @@ duckdb::string_t StringCast::Operation(timestamp_t input, Vector &vector) {
 	return result;
 }
 
+template <>
+duckdb::string_t StringCast::Operation(timestamp_ns_t input, Vector &vector) {
+	if (input == timestamp_t::infinity()) {
+		return StringVector::AddString(vector, Date::PINF);
+	} else if (input == timestamp_t::ninfinity()) {
+		return StringVector::AddString(vector, Date::NINF);
+	}
+
+	date_t date_entry;
+	dtime_t time_entry;
+	int32_t nanos;
+	Timestamp::Convert(input, date_entry, time_entry, nanos);
+
+	int32_t date[3], time[5];
+	Date::Convert(date_entry, date[0], date[1], date[2]);
+	Time::Convert(time_entry, time[0], time[1], time[2], time[3]);
+	// Use picoseconds so we have 6 digits
+	time[4] = nanos * 1000;
+
+	// format for timestamp is DATE TIME (separated by space)
+	idx_t year_length;
+	bool add_bc;
+	char micro_buffer[6];
+	char nano_buffer[6];
+	idx_t date_length = DateToStringCast::Length(date, year_length, add_bc);
+	idx_t time_length = TimeToStringCast::Length(time, micro_buffer);
+	idx_t nano_length = 0;
+	if (time[4]) {
+		//	If there are ps, we need all the µs
+		time_length = 15;
+		nano_length = 6;
+		nano_length -= NumericCast<idx_t>(TimeToStringCast::FormatMicros(time[4], nano_buffer));
+	}
+	idx_t length = date_length + time_length + nano_length + 1;
+
+	string_t result = StringVector::EmptyString(vector, length);
+	auto data = result.GetDataWriteable();
+
+	DateToStringCast::Format(data, date, year_length, add_bc);
+	data += date_length;
+	*data++ = ' ';
```

Comment:
can we add parenthesis here just for sanity?
I assume the order of operations is `*(data++)` ?

Reply:
Well this is an idiom that dates from the early days of the C language, so I think that adding parentheses would make it _more_ confusing because it would look like we are doing something strange when we are not!